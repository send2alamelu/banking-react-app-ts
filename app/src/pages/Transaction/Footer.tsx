import { useState } from 'react';
import { useHistory } from 'react-router-dom';

import BottomNavigationAction from '@mui/material/BottomNavigationAction';
import SendIcon from '@mui/icons-material/Send';
import CancelIcon from '@mui/icons-material/Cancel';

import BottomNav from '../../Components/Common/BottomNav/BottomNav';

export default function Footer() {
  const history = useHistory();

  const cancelTransaction = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();
    // TODO: Call API
    history.push("/dashboard");
  }

  const sendTransaction = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();
    // TODO: Call API
    history.push("/dashboard");
  }

  return (
    <BottomNav>
      <BottomNavigationAction label="Cancel" onClick={cancelTransaction} icon={<CancelIcon />} />
      <BottomNavigationAction label="Submit" onClick={sendTransaction} icon={<SendIcon />} />
    </BottomNav>
  );
}
