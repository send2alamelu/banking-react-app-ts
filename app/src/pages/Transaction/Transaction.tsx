import React from 'react';

function Transaction() {
  return (
    <div className="Transaction">
      Transaction Form Goes here...
    </div>
  );
}

export default Transaction;
