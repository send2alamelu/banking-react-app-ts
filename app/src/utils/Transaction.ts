import { TransactionItem } from "../types/TransactionItem";

const isTransfered = (type: string) => {
  return (type === 'transfer')
}

const isReceived = (type: string) => {
  return (type === 'receive')
}

export const getDescriptionFromTransactionType = (transactionItem: TransactionItem) => {
  if (isTransfered(transactionItem?.type)) {
    return `Transfered to ${transactionItem?.to?.accountHolderName}`
  } else if (isReceived(transactionItem?.type)) {
    return `Received from ${transactionItem?.from?.accountHolderName}`
  }

  return ''
}

export const currencyFormatter = (amount: number, currency: string = 'SGD') => {
  return new Intl.NumberFormat('en-SG', {
    style: 'currency',
    currency
  }).format(amount)
}