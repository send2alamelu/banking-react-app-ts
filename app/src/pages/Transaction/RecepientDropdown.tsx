import { useEffect, useState } from "react";
import Dropdown from "../../components/Fields/Dropdown/Dropdown";
import useGet from "../../hooks/useGet";
import { GetResponse } from "../../types/ApiResponse";
import { getPayeeApiPath } from "../../utils/Api";

interface Props {
  onChange: Function;
}

function RecepientDropdown(props: Props) {
  const getPayee: GetResponse = useGet(getPayeeApiPath());
  const [recepientList, setRecepientList] = useState([]);

  useEffect(() => {
    const recepients = getPayee?.response?.data || []
    if (recepients.length > 0) {
      setRecepientList(recepients.map(((recepient: any) => ({
        value: recepient.accountNo,
        label: recepient.accountHolderName,
      }))));
    }

  }, [getPayee?.response]);

  return (
    <Dropdown label="Recepient" items={recepientList} onChange={props.onChange} />
  )
}

export default RecepientDropdown