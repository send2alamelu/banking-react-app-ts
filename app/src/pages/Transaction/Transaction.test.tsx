import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Transaction from './Transaction';

describe('Transaction Component', () => {
  afterEach(() => {
    cleanup();
  });

  test('renders Transaction with children', () => {
    const { asFragment } = render(<Transaction />);
    expect(asFragment()).toMatchSnapshot();
  });
});