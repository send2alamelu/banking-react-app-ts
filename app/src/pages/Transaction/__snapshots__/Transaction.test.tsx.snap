// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Component renders Transaction with children 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-18j0g43"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Make a Transfer
    </h4>
  </div>
  <div
    class="MuiBox-root css-1l9hhr"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        id="demo-simple-select-label"
      >
        Recepient
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-label demo-simple-select"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="demo-simple-select"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-dhh0px"
          >
            <span>
              Recepient
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1l9hhr"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          Data of Transfer
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            aria-label="Choose date"
            aria-readonly="true"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            readonly=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-dhh0px"
            >
              <span>
                Data of Transfer
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1l9hhr"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="standard-basic"
          id="standard-basic-label"
        >
          Description
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="standard-basic"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1l9hhr"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="standard-basic"
          id="standard-basic-label"
        >
          Amount
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="standard-basic"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-xuhmcu"
  >
    <div
      class="MuiBottomNavigation-root css-16lloyr-MuiBottomNavigation-root"
    >
      <button
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CancelIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          Cancel
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SendIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          Submit
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
